pkg/declarativeconfig/permission_set.go

```diff
@@ -0,0 +1,33 @@
+package declarativeconfig
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"gopkg.in/yaml.v3"
+)
+
+type PermissionSet struct {
+	Name             string              `yaml:"name,omitempty"`
+	Description      string              ` yaml:"description,omitempty"`
+	ResourceToAccess map[string]DCAccess `yaml:"resource_to_access,omitempty"`
```

Comment:
This is more of a style thing rather than anything else, but:
Currently, if you would specify those resource, you would do:
```yaml
resource_to_access:
  Cluster: READ_ACCESS
  Namespace: READ_ACCESS
```

Making it more like look & feel compared to K8S resources, and looking forward if we were to do this within CRs, we would probably use an array instead of a map, as such:
```
resource_to_access:
- resource: Cluster
  access: READ_ACCESS
```
I personally find the latter one better in line with existing K8S resources + our own CR compared to a `map[string]string`.